sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/implementation/AzureActiveDirectoryConfiguration.java

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.aad.implementation;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.azure.spring.autoconfigure.aad.AADOAuth2AutoConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.core.AuthorizationGrantType;
+
+@Configuration
+@ConditionalOnClass(ClientRegistrationRepository.class)
+@EnableConfigurationProperties(AzureActiveDirectoryProperties.class)
+@AutoConfigureAfter(AADOAuth2AutoConfiguration.class)
+@ConditionalOnExpression("#{'${azure.active.directory.uri:notExist}' != 'notExist'}")
```

Comment:
Refs: https://stackoverflow.com/questions/49013862/how-to-get-spel-spring-el-default-to-null-expression-working-in-the-spring-xml